src/AppBundle/Command/DeleteUserCommand.php

```diff
@@ -0,0 +1,204 @@
+    }
+
+    /**
+     * This method is executed after initialize() and before execute(). Its purpose
+     * is to check if some of the options/arguments are missing and interactively
+     * ask the user for those values.
+     *
+     * This method is completely optional. If you are developing an internal console
+     * command, you probably should not implement this method because it requires
+     * quite a lot of work. However, if the command is meant to be used by external
+     * users, this method is a nice way to fall back and prevent errors.
+     */
+    protected function interact(InputInterface $input, OutputInterface $output)
+    {
+        if (null !== $input->getArgument('username')) {
+            return;
+        }
+
+        // multi-line messages can be displayed this way...
+        $output->writeln('');
+        $output->writeln('Delete User Command Interactive Wizard');
+        $output->writeln('-----------------------------------');
+
+        // ...but you can also pass an array of strings to the writeln() method
+        $output->writeln(array(
+            '',
+            'If you prefer to not use this interactive wizard, provide the',
+            'arguments required by this command as follows:',
+            '',
+            ' $ php app/console app:delete-user username',
+            '',
+        ));
+
+        $output->writeln(array(
+            '',
+            'Now we\'ll ask you for the value of all the missing command arguments.',
+            '',
+        ));
+
+        // See http://symfony.com/doc/current/components/console/helpers/questionhelper.html
+        $console = $this->getHelper('question');
+
+        $username = $input->getArgument('username');
+        if (null === $username) {
+            $question = new Question(' > <info>Username</info>: ');
+            $question->setValidator(array($this, 'usernameValidator'));
+            $question->setMaxAttempts(self::MAX_ATTEMPTS);
+
+            $username = $console->ask($input, $output, $question);
+            $input->setArgument('username', $username);
+        } else {
+            $output->writeln(' > <info>Username</info>: ' . $username);
+        }
+    }
+
+    /**
+     * This method is executed after interact() and initialize(). It usually
+     * contains the logic to execute to complete this command task.
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $startTime = microtime(true);
+
+        $username = $input->getArgument('username');
+
+        $this->usernameValidator($username);
+
+        $user = null;
+        $repository = $this->entityManager->getRepository('AppBundle:User');
+
+        $user = $repository->findOneBy(array('username' => $username));
+
+        if (null === $user) {
+            throw new \RuntimeException(sprintf('User with username "%s" not found.', $username));
+        }
+
+        $userId = $user->getId();
```

Comment:
@xabbuh, I think you're wrong. See the doctrine documentation: http://doctrine-orm.readthedocs.org/en/latest/reference/working-with-objects.html#removing-entities

```After an entity has been removed its in-memory state is the same as before the removal, except for generated identifiers.```